src/PsrContainerFakerProviderFactory.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CalebDW\Fakerstan;
+
+use Faker\Generator;
+
+class PsrContainerFakerProviderFactory
+{
+    private static string $phpContainerPath;
+    private static ?string $setsVariable;
+    private static string $containerFakerId;
+
+    public function __construct(
+        string $phpContainerPath,
+        ?string $setsVariable = null,
+        string $containerFakerId = Generator::class,
+    ) {
+        self::$phpContainerPath = $phpContainerPath;
+        self::$setsVariable = $setsVariable;
+        self::$containerFakerId = $containerFakerId;
+    }
```

Comment:
I completely agree on that. Honestly, that whole static variable stuff was done because when I saw `factory: @%fakerstan.fakerProviderFactory%::create` in the configuration file, I assumed that meant that it would have to be a static method on the class. And therefore require static variables, and so on :s